### DISCUSSION TITLE
Auto Drop stopped being reliable.

### DISCUSSION THREAD

**Discussion by JSTCOFFEE** [artisan-roaster-scope/artisan#1171] [Q&A]
I've been using Artisan (2.8.2 f525b16) in an automated fashion for months and months and recently the auto Drop function has started to be very unreliable. Last night I did a roast and the ET and BT crashed by 200 degrees after I dropped the beans but the Auto Drop was not triggered, even after waiting for a couple of minutes to make sure. I'm trying to figure out the cause; is the Auto Drop triggered off BT or ET or both? Is there anything I can do to belt and suspenders this?

**Comment by poundy** [comment]
In my experience, Autodrop is most sensitive to noise on probes. I did a DIY job on an uninstrumented roaster, and found the noise to be too much to allow for proper detection of autodrop, so went manual.  Have you seen any recent changes to hardware, probes, or the readings you see in Artisan?

**Comment by JSTCOFFEE** [comment]
I haven't changed the settings or probes for the ET & BT, but I did add an additional Extra Device. I've assumed Artisan looked at ET and or BT with Autodrop.

**Comment by MAKOMO** [comment]
Artisans autoDROP checks only the BT signal. Adding an ExtraDevice should not harm if this does not increase the noise level of the other channels somehow.

**Comment by JSTCOFFEE** [comment]
I have the heater power set to automatically go to zero after autoDROP is triggered. In this roast I dropped, then closed up the roaster, then opened it (I changed my mind as to what I was going to do) and you can see the heater doesn't get turned off until I did it manually a minute and a half after drop. In this roast I've already gone back and manually put in the drop time.

![image](https://github.com/artisan-roaster-scope/artisan/assets/111896828/8bbb0b41-3b08-4f3d-849a-bd4dc2e92d74)

**Reply by MAKOMO** [reply]
Could you please attach this and maybe one or two other profiles as .alog where AutoDROP did not work for you well (you need to drag and drop them here on GitHub on a comment field and before rename it to end with `.txt`. Thanks!

**Comment by roasterdave** [comment]
To help further we need to see your current settings and a profile file where the auto DROP did not trigger.  The profile you show a clip image from above would be good.  Use Help>Save Settings to write the current settings to a file.  Rename both the settings file (.aset) and the profile file (.alog) with .txt as the suffix.  Then drag them into a reply here on Github.  Do not attach them to an email reply.

**Comment by JSTCOFFEE** [comment]
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/11518419/artisan-settings.txt)
[DRU1-148_23-05-17_2109.txt](https://github.com/artisan-roaster-scope/artisan/files/11518421/DRU1-148_23-05-17_2109.txt)

**Reply by roasterdave** [reply]
The change in BT is very gentle at the point where you marked DROP.  It is still quite gentle later when you opened the roaster again.  Artisan looks for a sharper change in BT to detect autoDrop.  Let's compare to a curve where autoDROP marked correctly.  Would you post an older profile file where autoDROP marked correctly?  

Since you say nothing else changed, I suspect your BT probe has gotten dirty, which will slow down its response time and make your BT curve smoother.

**Comment by JSTCOFFEE** [comment]
When the autoDROP feature works as normal, like in this example, it is usually late by about 5-10 seconds and then I will go in manually and fine tune the actual time. While none of my settings have changed between these two roasts, my roasting has changed. What I usually do with the probe once I drop the beans is just rest it on the bottom of the roaster but I think what I can do is install an insulated rest to help make the autoDROP more reliable. Making the 'aggressiveness' or 'window' of the autoCHARGE and autoDROP features would probably be helpful for the DIYers, not sure if it would be worthwhile for you to code though.

[DRU1-106_22-11-20_1511.txt](https://github.com/artisan-roaster-scope/artisan/files/11519101/DRU1-106_22-11-20_1511.txt)

**Comment by roasterdave** [comment]
Again a very gentle slope.  This one may be just detectable.  Your setup, moving the probe, is not what autoDROP was modeled after.  If you cause a sharper change in BT autoDROP will work and the timing will improve.  

Artisan is open source so you can modify the autoDROP alogrithm to fit your circumstance.  If you create a better general solution you can generate a PR (pull request).
